### DISCUSSION TITLE
add sidenotes inside an equation

### DISCUSSION THREAD

**Discussion by EEGStudy** [fmarotta/kaobook#208] [Q&A]
Hello! I am a newbie to Kaobook, but started to use it right away.  I added a sidenote inside a simple equation, but Overleaf cannot compile the code, complaining that "Not in outer par mode" and told me "You'ave lost some text. Try typing <return> to proceed"

Help will be greatly appreciated.

BTW, the manual says (page 2): 

"The commands \frontmatter, \mainmatter and \backmatter have been redefined in order to have automatically wide margins in
the main matter, and narrow margins in the front and back matters. However, the page style can be changed at any moment, even in
the middle of the document."

Can I change the page layout, or just the (header) style, **in the middle of the document**? I have a project where at the bottom half of a page, I want to use the wide page to get more space.  So I tried to change the layout from Margin to Wide in the middle of the page, but it did not work.  Any hint? 

Thanks a lot!

**Comment by fmarotta** [comment]
Hi, regarding the page layout, have you tried the `widepar` environment? Usage is as follows:
```
\begin{widepar}
<my wide content>
<...>
\end{widepar}
```
If you have an older version of kaobook, you may also use the `addmargin` environment (read more about it in the [KOMAScript documentation](https://ctan.ebinger.cc/tex-archive/macros/latex/contrib/koma-script/doc/scrguien.pdf). kaobook's `widepar` is just a wrapper around `addmargin`. In particular, it is defined as follows:
```
\NewDocumentEnvironment{widepar}{}{%
    \if@twoside%
    \Ifthispageodd{%
        \begin{addmargin}[0cm]{-\marginparwidth-\marginparsep}%
    }{%
        \begin{addmargin}[-\marginparwidth-\marginparsep]{0cm}%
    }%
    \else%
    \begin{addmargin}[0cm]{-\marginparwidth-\marginparsep}%
    \fi%
}{%
    \end{addmargin}%
}
```
Feel free to copy this definition in your document, in case your version of kaobook is so old that it doesn't have the `widepar` environment.

Changing the header should also be possible. The recommended way is to create a new page style in the preamble of your document with the `\newpagestyle` command, and then calling `\pagestyle{<mynewpagestyle>}` in the middle of the document whenever you want to use your new page style. The syntax for creating a new page style is
```
\newpagestyle{name }{header specification }{footer specification }
```
where a "specification" is
```
(length of the line above ,thickness of the line above )%
{specification for the left page in two-side layout }%
{specification for the right page in two-side layout }%
{specification for all pages in one-side laypout }%
(length of the line below ,thickness of the line below )
```
You can find some examples in `kao.sty` at lines 230 and following.

Regarding the sidenote inside an equation, I would try to pass an offset of `0pt` to `\sidenote` (check the documentation for the correct syntax when you want to pass an offset to `\sidenote`). This would convert the sidenote from floating to not floating, and might solve your problem. Let me know how it goes.
